CustomCollections/CustomCollections/CustomVector.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CustomCollections
+{
+    class CustomVector<T> : IList<T>
+    {
+        private T[] _vector;
+
+        public int Count { get; private set; }
+        public bool IsReadOnly { get; private set; } = true;
+
+        public CustomVector()
+        {
+            _vector = new T[0];
+            Count = _vector.Length;
+        }
+
+        public T this[int index]
+        {
+            get => _vector[index];
+            set => _vector[index] = value;
+        }
+
+        public void Add(T item)
+        {
+            var newVector = new T[Count + 1];
+            for (int i = 0; i < Count; i++)
+            {
+                newVector[i] = _vector[i];
+            }
+            newVector[Count] = item;
+            Count++;
+            _vector = newVector;
+            IsReadOnly = false;
+        }
+
+        public void Clear()
+        {
+            _vector = new T[0];
+            Count = 0;
+            IsReadOnly = true;
+        }
+
+        public bool Contains(T item)
+        {
+            if (_vector.Contains(item))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public bool Remove(T item)
+        {
+            _vector = _vector.Where(s => !s.Equals(item)).ToArray();
+            if (Count > _vector.Length)
+            {
+                Count = _vector.Length;
+                return true;
+            }
+            return false;
+        }
+
+        public void RemoveAt(int index)
+        {
+            Remove(_vector[index]);
+        }
+
+        public void Insert(int index, T item)
+        {
+            Count++;
+            var newVector = new T[Count];
+            var j = 0;
+            for (int i = 0; i < Count; i++)
+            {
+                if (i == index)
+                {
+                    newVector[i] = item;
+                }
+                else
+                {
+                    newVector[i] = _vector[j];
+                    j++;
+                }
+            }
+            _vector = newVector;
+        }
+
+        public int IndexOf(T item)
+        {
+            for (int i = 0; i < Count; i++)
+            {
+                if (_vector[i].Equals(item))
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            if (array.Length - arrayIndex < Count)
+            {
+                Console.WriteLine("You can't do this. The array is too small or arrayIndex is in the end of array");
```

Comment:
This should throw exception, not write something to Console. You can watch list of expected exceptions in documentation page of interface methods. Refer to :https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1.copyto?view=netframework-4.7.2#System_Collections_Generic_ICollection_1_CopyTo__0___System_Int32_